File: Java/IntegerListFromUsp/src/Exercicio06.java
import java.util.Scanner;

public class Exercicio06 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        scanner.reset();
        int maiorNota = 0;
        int menorNota = 100;
        System.out.print("Digite a quantidade de aluno: ");
        int numeroAlunos = scanner.nextInt();
        int nota = 0;
        for (int i = 0; i < numeroAlunos; i++) {
            System.out.print("Nota %d: ".formatted(i + 1));
            nota = scanner.nextInt();
            if (nota > maiorNota) {
                maiorNota = nota;
            }
            if (nota < menorNota) {
                menorNota = nota;
            }
        }

        System.out.println("A maior nota é: %d".formatted(maiorNota));
        System.out.println("A menor nota é: %d".formatted(menorNota));
    }
}

File: CPP/IntegerListFromUsp/src/Exercicio06.cpp
#include <iostream>

using namespace std;

int main() {
    int n;
    cout << "Quantidade de alunos: ";
    cin >> n;

    int maior = -1;
    int menor = 101;

    for (int i = 1; i <= n; i++) {
        int nota;
        cout << "Nota do aluno " << i << ": ";
        cin >> nota;
        if (nota < 0 || nota > 100) {
            cout << "Nota inválida (0 a 100)." << endl;
            i--;
            continue;
        }
        if (nota > maior) maior = nota;
        if (nota < menor) menor = nota;
    }

    cout << "Maior nota: " << maior << endl;
    cout << "Menor nota: " << menor << endl;
    return 0;
}
